### DISCUSSION TITLE
class path extraction

### DISCUSSION THREAD

**Discussion by horakhsh** [classgraph/classgraph#512] [Q&A]
Hi all
I am using ClassGraph to extract class path in a web application(war) that runs on Wildfly app server with the following code : 

new ClassGraph()
                .verbose(true)
                .ignoreParentClassLoaders()
                .overrideClassLoaders(Thread.currentThread()
                                                                 .getContextClassLoader())
                .getClasspath()

In the returned class path string, in addition to class path of this deployment, there is unexpected paths from other deployments. In a web environment, to avoid class conflict, we expect that the class path of each deployed application be separate from the class path of other deployed application. This is n't the case with ClassGraph. why ? 

with best regards

**Comment by lukehutch** [comment]
Hi @horakhsh,

Are you able to attach the logs, please? It's hard to tell without seeing the logs. Thanks!

**Comment by lukehutch** [comment]
Hi @horakhsh -- just following up on this. Can you please provide logs? Even better, can you please provide a small reproducer project with two different simple deployments, where one deployment can see the classes from the other deployment? I have no idea how to set that up with Wildfly.
